### PR TITLE
check if ceph is installed

### PR DESCRIPTION
**Pull request by alfredodeza**
This is a PR with a bit of feature creep: the intent was to use the new `Ceph` object to provide an interface for versions and use it to map it against supported `systemd` versions.

It is not entirely sure if this approach will be used, but for now this can be used to fix (http://tracker.ceph.com/issues/11115) which allows to check if Ceph is installed and error accordingly. 

The "check if ceph is installed" comes free with this implementation, if we decide later to not use versions we can change the approach.

This PR also moves the private function in `hosts/__init__.py` to `util.versions` because the functionality is now being used by more than one place. At some point the tests should map to the new location - but all tests pass as it is now.

### CODE REVIEW

**Comment by trhoden** [comment]
The main thing that comes to mind with this is if there is a way we can keep from doing the check for if Ceph is installed over and over again...

I realize that sometimes we want it, sometimes we don't.

What if it becomes an optional argument to `hosts.get()`?  When we are initiating the connection with `hosts.get()`, we know whether we require to already be installed or not, so maybe adding it in by doing:

`hosts.get(..., ceph_required=True)` ?

Just thinking out loud.

File: ceph_deploy/hosts/__init__.py [hunk 1]
```diff
@@ -70,6 +76,10 @@ def get(hostname,
     module.machine_type = machine_type
     module.init = module.choose_init(module)
     module.packager = module.get_packager(module)
+    # execute each callback if any
+    if callbacks:
+        for c in callbacks:
+            c(module)
```
Comment (trhoden) [inline ceph_deploy/hosts/__init__.py]: Awesome!

File: ceph_deploy/mon.py [hunk 2]
```diff
@@ -257,7 +261,7 @@ def mon_create(args):
         try:
             # TODO add_bootstrap_peer_hint
             LOG.debug('detecting platform for host %s ...', name)
-            distro = hosts.get(host, username=args.username)
+            distro = hosts.get(host, username=args.username, callbacks=[packages.ceph_is_installed])
```
Comment (trhoden) [inline ceph_deploy/mon.py]: This looks like the only `hosts.get()` call that didn't get split into multiple lines...  Not sure I care. :)

Comment (alfredodeza) [inline ceph_deploy/mon.py]: I think it wasn't long enough. Not a problem if you prefer it split.

**Pull request merged**